docs/enhancements/dual-stack-vips.md

```diff
@@ -0,0 +1,104 @@
+
+### API Extensions
+
+New fields will have to be added to the Cluster object to persist the specified VIPs.
+
+### Implementation Details
+
+#### Phase 1
+
+`cluster-create-params`, `v2-cluster-update-params` and `cluster` will get new fields
+
+* `api_vips` of type `array`
+* `ingress_vips` of type `array`
+
+Respective AgentClusterInstall CR will also get those new fields.
+
+The database schema will be updated by creating new tables `api_vips` and `ingress_vips` with the following columns
+
+* `cluster_id`
+* `vip`
+
+#### Phase 2
+
+A set of functions for cluster creation and cluster update will provide a compatiblity between already existing fields (`api_vip` and `ingress_vip`) and new ones (`api_vips` and `ingress_vips`). The behaviour will be as follows
+
+* If only `api_vip` is provided, the `api_vips` will be automatically populated by a simple conversion from `string` to `[]string`. This guarantees no change in behaviour for users not aware of introduction of the new field.
+
+* If only `api_vips` are provided, the `api_vip` will be populated by taking the first value from `[]string`. This guarantees that older versions of the UI do not break when interacting with the new field.
+
+* If both `api_vip` and `api_vips` are provided, the value of `api_vips` will take precedence. This is an undesired behaviour indicating use of the wrong syntax or client but for the completeness the behaviour has to be defined.
+
+All the points described above will behave in an equivalent way for `ingress_vip` and `ingress_vips`.
+
+#### Phase 3
+
+After providing a required period of backwards-compatibility the `api_vip` and `ingress_vip` fields will be deprecated and removed. The value in the database will be nulled. Removal of the column in the database will follow.
+
+### Risks and Mitigations
+
+The similar manouever of introducing `[]string` in place of `string` has already happened in the past when implementing the initial dual-stack support. An example of the field that got transformed is `machine_network_cidr` that became `machine_networks`.
+
+There is already an experience and established path, therefore the risk is considered minimal.
+
+Given that the core of the feature is implemented in the baremetal networking component of the OCP, the implementation in Assisted Installer is only responsible for providing the correct syntax to the install-config.
+
+### UI Impact
+
+UI changes will be required in order to
+
+* display details of dual-stack clusters with dual-stack VIPs correctly
+* allow configuring dual-stack VIPs for dual-stack clusters
+
+For displaying the changes, the UI will need to display the value of `api_vips` and `ingress_vips` instead of the current `api_vip` and `ingress_vip`.
+
+For setting the values, the UI will need to include `api_vips` and `ingress_vips` in the cluster payload. The previous fields i.e. `api_vip` and `ingress_vip` must not be provided.
```

Comment:
Note that this differs from the [k8s API change guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#making-a-singular-field-plural), which says a new client must supply both values.

Reply:
Agree, that’s a good catch. Seems like what we did in the past was not in line with k8s guidelines. I will of course fix it here so we are following them now

Reply:
Fixed, I have rewritten the above to follow the API change guide